### PR TITLE
Create a role in the specified Org or current org in use

### PR DESCRIPTION
**Pull request by sahithi**
Inspected code as per pep8 standards.

### CODE REVIEW

File: pyvcloud/vcd/org.py [hunk 1]
```diff
@@ -77,6 +77,32 @@ def create_catalog(self, name, description):
             EntityType.ADMIN_CATALOG.value,
             catalog)
 
+    def create_role(self, role_name, description, rights):
+        """
+        Creates a role in the organization
+        :param role_name: name of the role to be created
```
Comment (anusuyar) [inline pyvcloud/vcd/org.py]: Let's also specify the type of each argument. 
like 
:param role_name:  (str): name of the role to be created

File: pyvcloud/vcd/org.py [hunk 2]
```diff
@@ -582,6 +608,39 @@ def get_roles_query(self, name_filter=None):
...
+        """  # NOQA
+        if self.resource is None:
+            self.resource = self.client.get_resource(self.href)
+
+        resource_type = 'right'
+        query = self.client.get_typed_query(
+            resource_type,
+            query_result_format=QueryResultFormat.RECORDS,
+            equality_filter=name_filter)
+        return query, resource_type
```
Comment (anusuyar) [inline pyvcloud/vcd/org.py]: Why are we passing resource_type here? I dont see it being used in the calling function.

File: tests/vcd_role.py [hunk 3]
```diff
@@ -26,5 +26,13 @@ def test_get_rights(self):
         rights = role.list_rights()
         assert len(rights) > 0
 
+    def test_create_role(self):
```
Comment (anusuyar) [inline tests/vcd_role.py]: We can add copyrights header for test files too.

File: pyvcloud/vcd/org.py (hunk 2 above)
Comment (sahithi) [reply pyvcloud/vcd/org.py]: It is going to be used in another change-list underway (list rights in the org).

File: pyvcloud/vcd/org.py [hunk 4]
```diff
@@ -77,6 +77,32 @@ def create_catalog(self, name, description):
...
+        :param rights: names of zero or more rights to be associated with the role
+        :return: RoleType just created
+        """  # NOQA
+        org_admin_resource = self.client.get_resource(self.href_admin)
+        role = E.Role(
+            E.Description(description),
+            E.RightReferences(),
+            name=role_name
+        )
+        for right in rights:
```
Comment (rocknes) [inline pyvcloud/vcd/org.py]: Before iterating, I think we should make sure that rights is a list. Also a Null check should be made.

File: tests/vcd_role.py (hunk 3 above)
Comment (sahithi) [reply tests/vcd_role.py]: fixed.

File: pyvcloud/vcd/org.py [hunk 5]
```diff
@@ -582,6 +608,39 @@ def get_roles_query(self, name_filter=None):
...
+        :return: (QueryResultRightRecordType): Right query result in records
+                 format
+        """  # NOQA
+        try:
+            rights_query = self.get_rights_query(('name', right_name))[0]
+            return rights_query.find_unique()
+        except MissingRecordException:
+            raise Exception('Right \'%s\' does not exist.' % right_name)
+
+    def get_rights_query(self, name_filter=None):
```
Comment (rocknes) [inline pyvcloud/vcd/org.py]: This looks out of place. Query related stuff should be in client.py

File: pyvcloud/vcd/org.py (hunk 2 above)
Comment (rocknes) [reply pyvcloud/vcd/org.py]: I don't think we should be passing back resource_type unless we make get_rights_query a generic function. The name of the function is a dead giveaway that we are dealing with 'rights'.

File: tests/vcd_role.py [hunk 6]
```diff
@@ -26,5 +41,13 @@ def test_get_rights(self):
...
+        logged_in_org = self.client.get_org()
+        org = Org(self.client, resource=logged_in_org)
+        role_name = self.config['vcd']['role_name']
+        org.create_role(role_name, 'test role', {'vApp: Copy'})
```
Comment (pacogomez) [inline tests/vcd_role.py]: If the method is expecting a tuple of strings, this should be `, ('vApp: Copy'))`, not a dict

File: pyvcloud/vcd/org.py (hunk 5 above)
Comment (pacogomez) [reply pyvcloud/vcd/org.py]: We have `list_*`  methods outside client.py. Maybe this should renamed to `list_rights`?

File: pyvcloud/vcd/org.py (hunk 2 above)
Comment (pacogomez) [reply pyvcloud/vcd/org.py]: This should only return one value, the list of rights, see `list_catalogs` as an example

File: tests/vcd_role.py (hunk 6 above)
Comment (sahithi) [reply tests/vcd_role.py]: my mistake, fixed.

File: pyvcloud/vcd/org.py (hunk 4 above)
Comment (sahithi) [reply pyvcloud/vcd/org.py]: I changed:  for right in rights -> for right in tuple(rights). This doesn't require None check.

File: pyvcloud/vcd/org.py [hunk 7]
```diff
@@ -582,6 +610,39 @@ def get_roles_query(self, name_filter=None):
...
+        """  # NOQA
+        right_record = self.list_rights(('name', right_name))
+        if len(right_record) < 1:
+            raise Exception('Right \'%s\' does not exist.' % right_name)
+        return right_record[0]
+
+    def list_rights(self, name_filter=None):
+        """
+        Get the typed query for the rights in the current Org
+        :param name_filter: (tuple): (name ,'right name') Filter the rights by
```
Comment (rocknes) [inline pyvcloud/vcd/org.py]: I don't think we need a tuple here, rather a support for multiple filters (by name) which can be logically or-ed will be more valuable. 

fyi : Rights are unique in context of an org, so this list will always contain 1 or 0.

File: tests/vcd_role.py [hunk 8]
```diff
@@ -26,5 +41,13 @@ def test_get_rights(self):
         rights = role.list_rights()
         assert len(rights) > 0
 
+    def test_create_role(self):
+        logged_in_org = self.client.get_org()
+        org = Org(self.client, resource=logged_in_org)
+        role_name = self.config['vcd']['role_name']
+        org.create_role(role_name, 'test description', ('Disk: View Properties',))
+        role_record = org.get_role(role_name)
+        assert self.config['vcd']['role_name'] == role_record.get('name')
```
Comment (rocknes) [inline tests/vcd_role.py]: Please make sure you clean up the role after the tests are done.

Comment (sahithi) [reply tests/vcd_role.py]: Yes, I will make a not to modify this test once I implement the command to delete a role.

**Pull request merged**